src/ConductorSharp.Engine/Builders/Configurable/WorkflowDefinitionBuilder.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Autofac;
+using ConductorSharp.Client.Model.Common;
+using ConductorSharp.Engine.Interface;
+using ConductorSharp.Engine.Util;
+using ConductorSharp.Engine.Util.Builders;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace ConductorSharp.Engine.Builders.Configurable
+{
+    public static class WorkflowOutputExtension
+    {
+        public static void SetOutput<FWorkflow, F, G>(this WorkflowDefinitionBuilder<FWorkflow, F, G> builder, Expression<Func<FWorkflow, G>> input)
+            where FWorkflow : Workflow<FWorkflow, F, G>
+            where F : WorkflowInput<G>
+            where G : WorkflowOutput
+        {
+            builder.BuildContext.Outputs = ExpressionUtil.ParseToParameters(input.Body);
+        }
+
+        public static void SetOptions<FWorkflow, F, G>(this WorkflowDefinitionBuilder<FWorkflow, F, G> builder, Action<WorkflowOptions> adjustOptions)
+            where FWorkflow : Workflow<FWorkflow, F, G>
+            where F : WorkflowInput<G>
+            where G : WorkflowOutput
+        {
+            adjustOptions?.Invoke(builder.BuildContext.WorkflowOptions);
+        }
+    }
+
+    public class WorkflowDefinitionBuilder<TWorkflow, TInput, TOutput>
+        where TWorkflow : ITypedWorkflow
+        where TInput : WorkflowInput<TOutput>
+        where TOutput : WorkflowOutput
+    {
+        private readonly Type _workflowType = typeof(TWorkflow);
+        private string _name;
+
+        public BuildContext BuildContext { get; } = new();
+        public BuildConfiguration BuildConfiguration { get; set; }
+        public WorkflowBuildItemRegistry WorkflowBuildRegistry { get; }
```

Comment:
Are there any cases where someone outside this assembly would use this property? If not make it `internal`

Reply:
Any builder extension should be able to access this this. The idea is to use this for cases where workflow definitions create some artifacts which are used elsewhere.